### PR TITLE
Add built-in function oct (#2115)

### PR DESCRIPTION
**Pull request by lawrshen**
Signed-off-by: lawrshen <191220090@smail.nju.edu.cn>

**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #2115 

**What this PR does / why we need it:**

Add a feature of oct() built-in function.

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: pkg/builtin/unary/oct.go [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+			Fn: func(origVec *vector.Vector, proc *process.Process, _ bool) (*vector.Vector, error) {
+				col := origVec.Col.([]int16)
+				results := &types.Bytes{
+					Data:    []byte{},
+					Offsets: make([]uint32, len(col)),
+					Lengths: make([]uint32, len(col)),
+				}
+				results = oct.OctInt16(col, results)
+
+				resVec, err := process.Get(proc, int64(len(results.Data)), types.Type{Oid: types.T_varchar, Size: 24})
```
Comment (broccoliSpicy) [inline pkg/builtin/unary/oct.go:312]: is the results.Data's length fixed? I don't know, for int16 ---> (16 % 3 + 1) * numberOfRows?

if we find a way to allocate the right size of memory for result.Data field, then we can get rid of the append(result.Data).... code later in our function execution, I think get rid of this append thing may help boost this function's performance

File: pkg/vectorize/oct/oct.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	OctInt16 = oct[int16]
+	OctInt32 = oct[int32]
+	OctInt64 = oct[int64]
+}
+
+func oct[T constraints.Unsigned | constraints.Signed](xs []T, rs *types.Bytes) *types.Bytes {
+	var cursor uint32
+
+	for idx := range xs {
+		octbytes := []byte(strconv.FormatUint(uint64(xs[idx]), 8))
```
Comment (broccoliSpicy) [inline pkg/vectorize/oct/oct.go:50]: I stated a few reasons why we may prefer using bit operation in lieu of the strconv.FormatUint function in the "bin" pull request, care to check it out

**Review by broccoliSpicy — CHANGES_REQUESTED** [review]
your code is neat, sorry if our requirement for performance seem draconian to you...

File: pkg/builtin/unary/oct.go (hunk 1 above)
Comment (lawrshen) [reply pkg/builtin/unary/oct.go:312]: I think the length may not be fixed. For example, a negative integer is expected to convert into a 64bit number like 
```MySQL
mysql> SELECT OCT(-128);
        -> '1777777777777777777600'
```
in MySQL. However positive integer is flexible.

I also approve that append(result.Data).... code is not the most efficient approach. A potential way is supposed to use 22*numberOfRows where 22 is the longest width for an integer. It wastes some space but we can return Data[:offset[last]+length[last]]. Traversing the original integers to judge length I think may not high-efficiency.

File: pkg/vectorize/oct/oct.go (hunk 2 above)
Comment (lawrshen) [reply pkg/vectorize/oct/oct.go:50]: Okay, I would to see it and make changes.

Comment (lawrshen) [reply pkg/vectorize/oct/oct.go]: fix it

**Review by broccoliSpicy — APPROVED** [review]

**Pull request merged**